### ISSUE TITLE
Client crash when entering Mystcraft Age w/ CraftPresence installed (1.7.10, possibly later versions?)

### ISSUE THREAD

**Issue by nan0bug00** [XCompWiz/Mystcraft-Issues#342]
Craftpresence ver: CraftPresence-1.7.10-1.4.9.9.jar
Mystcraft ver: mystcraft-1.7.10-0.12.3.04

I think the problem is either due to the way Mystcraft generates new dimensions (or reports their names?) , or how CraftPresence detects dimensions.  Not entirely sure which.  However the crash is resolved by disabling CraftPresence.  

The error

`java.lang.IllegalArgumentException: Could not get provider type for dimension 1210950779, does not exist`

references the IDs listed in Mystcraft's core.cfg on lines 37 and 38:

```
    I:ids.dim_provider=1210950779
    I:ids.villager.archivist=1210950779
```

The dimension ID that my player was entering was 2, with the name "Random Age", but the crash references this dim_provider ID. 

I'm aware that the 1.7.10 version of Mystcraft is no longer being updated, but wanted to report this in case it is still an issue in recent versions.  I have also reported this on the CraftPresence issue tracker [here](https://gitlab.com/CDAGaming/CraftPresence/issues/17) in case you would like to figure out a solution with the dev(s?).

[crash-2019-03-10_10.36.03-client.txt](https://www.hastebin.com/ahacucutun.sql)

**Comment by Jonius7** [comment]
Thank god I found this, it was crashing with provider -777 (Twilight Forest) then I changed it, and it crashed on 1210950779.
It seems to only accept the dimension provider with the same number as the dimension (ie. for Twilight Forest, 7), because my Mystcraft dimensions have Twilight Forest biomes in them, I guess.

**Comment by CDAGaming** [comment]
This Issue will be fixed in v1.5.2 of CraftPresence. Both of your guys' Issues were caused from an error I made in registering Dimensions that effected the mod in 1.8.9 and below.

The Method that I use had now been changed and v1.5.2 will be hopefully releasing on the 29th (I've had to delay it a few times as there are a LOT of other fixes in v1.5.2 and QoL Changes too.

CC: @nan0bug00 and @Jonius7 

This Issue has also been posted here: https://gitlab.com/CDAGaming/CraftPresence/issues/17

Feel free to head there too, as I posted an earlier fixed build while you wait for v1.5.2's official release.

**Comment by CDAGaming** [comment]
v1.5.2 has now been released, and this Issue can now be closed as Fixed 👍

**Comment by Veovis-Muaddib** [comment]
Nice, thanks for the fix @CDAGaming!

**Comment by Jonius7** [comment]
After testing the new v1.5.2 for CraftPresence, it is working, but rather a new problem has appeared. Mystcraft is constantly spamming "Someone is trying to get the age name from an improperly constructed world provider with dim id 0", and rather more importantly, it is causing significant frame drops (often dropping below 10 frames per second on my mod pack). I might test these two in isolation but it is happening as I am in the Overworld.

![image](https://user-images.githubusercontent.com/1049415/54917236-12e85880-4f47-11e9-8ea1-c36d10a92db6.png)

**Comment by CDAGaming** [comment]
Hey Jonius, I looked into it and it seems I missed an additional if case to prevent Infinite checking of dimension name as the class name and the actual dimension name CAN be different.

I'll roll out a hot fix sometime this week for v1.5.3 but for now all you need to do to prevent an infinite lookup is to just add a new value in dimension name's in which the value bane is the same as the current dimension name, which can be found via the /cp view currentData command.

**Comment by CDAGaming** [comment]
Update: v1.5.3 has a possibility of being delayed to the first two weeks of April. I found a number of Issues relating to the new Per-GUI Reimplementation that I'm working on fixing, then getting to this. Stay Tuned...

**Comment by CDAGaming** [comment]
@Jonius7 This should now be fixed in v1.5.3, that is now released. Let me know if it is/isn't 👍

**Comment by Jonius7** [comment]
Bug seems to be fixed and everything is working fine. Thanks!
